### PR TITLE
fix mysql_wsrep_stats

### PR DESCRIPTION
**Pull request by xinity**
a PR that fixes : https://github.com/collectd/collectd/pull/850

### CODE REVIEW

**Comment by rubenk** [comment]
Thanks. Could you also address the comment @mfournier made in the original PR at https://github.com/collectd/collectd/pull/850#discussion_r23298223

**Comment by xinity** [comment]
hi @rubenk 

actually i did adressed @mfournier comment :)

feel free to reply :)

by the way, don't merge there is more to come on this PR :)

**Comment by xinity** [comment]
this PR can now be merged :)
added features to monitoring a galera cluster using grafana :)

**Comment by rubenk** [comment]
I don't see that @mfournier's comment is addressed. When some items are not found, they would be submitted as empty values.

File: src/collectd.conf.in [hunk 1]
```diff
@@ -740,6 +740,12 @@
 #		SlaveStats true
 #		SlaveNotifications true
 #	</Database>
+#	<Database galera>
```
Comment (xinity) [inline src/collectd.conf.in]: @rubenk isn't it what was needed ?

**Comment by xinity** [comment]
@mfournier asked this:
"@XANi, could you please also add a WsrepStats example line to src/collectd.conf.in ?"

and that's what i did no ? 
@rubenk  see comment i've added in the commit for src/collectd.conf.in :)

there is more to come, but will be pushed through another PR

**Comment by rubenk** [comment]
That's not what I meant. The comment I linked to was:

> I might be mistaken, but wouldn't the code below submit empty values in the case one or the other of the items matched above aren't found ?
> 
> Wouldn't it be an idea to move these into *_submit() calls into the if/else clauses above, and avoid these numerous wsrepo_* variables in the process ?
> 
> It would make this consitent with how mysql_read() works and behaves at least, wouldn't it ?

**Comment by xinity** [comment]
oups sorry :(

i'll take a look at this ASAP


Le lun. 1 août 2016 à 10:03, Ruben Kerkhof <notifications@github.com> a
écrit :

**Comment by rubenk** [comment]
Not sure what went wrong here, but your last commit changes 212 files. Did you rebase your master branch by any chance?

**Comment by rubenk** [comment]
Much better, thank you.

One last thing, could you change your commit message to something more descriptive?
I suggest `mysql plugin: add support for Galera statistics`. That would make it more obvious as to what it does.

**Comment by xinity** [comment]
fixed and rewritten the function :)

sorry for the mess, been knocking about bad habits with git :(

**Comment by rubenk** [comment]
No, not fixed :) The commit message is now 'fix wsrep statistics'.

Just a tip, check your changes with git diff and git log before you push ;)

**Comment by xinity** [comment]
damned ! finally good right ?

i'll take cautious with my next PR :)

**Comment by rubenk** [comment]
You've added a single type in types.db, but the type_instances you use differ in type.
I'm not a 100% sure if this is allowed. https://collectd.org/wiki/index.php/Naming_schema#Plugin_instance_and_type_instance seems to suggest it is not. CC'ing @octo since he'll know best.

**Comment by xinity** [comment]
i'll wait for @octo feedback then :)
thanks a lot

**Comment by XANi** [comment]
It could easily use existing types (operations/total_bytes/queue_length), see my original patch: https://github.com/XANi/collectd/commit/9be0076f756ce6ea541804278d8dc820bbcb6f0f

Only extra type that was needed was `window_size`

**Comment by rubenk** [comment]
You are not using separate types, you're using separate type_instances for a single type 'mysql_wsrep'

**Comment by XANi** [comment]
@rubenk my patch used types to group similiar ops + type instances

so type bytes had instances wsrep_received and wsrep_replicated, type queue had wsrep_local_recv and wsrep_local_send etc.

Then I just grouped it by types on graphs like that : 
zaznaczenie_2016-08-04_001

And yes, you are right about types, you can't use same type name for gauge and counter

**Comment by rubenk** [comment]
Ok, can you rework the patch then?

**Comment by XANi** [comment]
@rubenk I could, I just dont have anything to actually compile and test it on (our wsrep cluster still runs 5.4.x with my old patch and I dont know if current head will even compile on centos 6...)

**Comment by rubenk** [comment]
Oh, ok, neither do we, so it would be good to get that sorted first.

Do you mind if I close this and you open a new PR when you're ready?

**Comment by XANi** [comment]
@rubenk I am not author of this pull request, maybe wait for him :)  I am author of one it was based of (https://github.com/collectd/collectd/pull/850), but after waiting months for it I gave up and just applied patch to our local version.

**Comment by octo** [comment]
> You are not using separate types, you're using separate type_instances for a single type 'mysql_wsrep'

+1

**Comment by rubenk** [comment]
@XANi I totally missed I was talking to two different persons, my apologies.

In any case, I'll close this, @xinity please open a new PR taking our feedback into account.

**Comment by xinity** [comment]
@rubenk i'm not sure  i get what you want me to fix, but yes, i'd be happy to fix it ASAP :)

**Comment by rubenk** [comment]
@xinity You'll need to use separate types, since a single type won't work.

**Comment by XANi** [comment]
@xinity give mie a sec and I'll try to sort it out

**Comment by xinity** [comment]
@XANi i'll wait then thanks for you help guys !

**Comment by octo** [comment]
@xinity You're using one *type*, `mysql_wsrep` for all sorts of different metrics, bytes, queue lengths, operations, … [This comment](https://github.com/collectd/collectd/pull/1797#issuecomment-237529093) sounds like it's going in the right direction.

Thanks for taking a look @XANi!

**Comment by xinity** [comment]
@octo got it , 
@XANi by looking at your PR, i see what's wrong with mine

**Comment by xinity** [comment]
i got it @octo 
@XANi let me know if you found something :)

**Comment by XANi** [comment]
@xinity https://github.com/XANi/collectd/tree/mysql-galera-state here is what I meant (all i can say is that it compiles, haven't tested it anywhere yet)

**Comment by xinity** [comment]
@XANi nice, i see :) let me fix and extended a bit my PR :) thanks a lot !

**Pull request closed without merging**